### PR TITLE
Test WrappedFunction properties

### PR DESCRIPTION
**Pull request by legendecas**
Add test coverages on [CopyNameAndLength](https://tc39.es/proposal-shadowrealm/#sec-copynameandlength) in [WrappedFunctionCreate](https://tc39.es/proposal-shadowrealm/#sec-wrappedfunctioncreate).

### CODE REVIEW

**Review by leobalter — APPROVED** [review]

File: test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+function fn() {}
+Object.defineProperty(fn, 'length', {
+  get: () => {
+    throw new Error('blah');
+  },
+  enumerable: false,
+  configurable: true,
+});
+fn;
+`), 'expect a TypeError on length getter throwing');
```
Comment (rwaldron) [inline test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js:36]: I think this should be: 

```js
assert.throws(TypeError, () => {
  r.evaluate(`
    function fn() {}
    Object.defineProperty(fn, 'length', {
      get: () => {
        throw new Error('blah');
      },
      enumerable: false,
      configurable: true,
    });
    fn;
  `)
}, 'expect a TypeError on length getter throwing');
```

And same for the test further down

**Review by rwaldron — CHANGES_REQUESTED** [review]

File: test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js (hunk 1 above)
Comment (rwaldron) [reply test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js:36]: Oh wait... NO, I'm wrong. This is awesome.

**Review by rwaldron — APPROVED** [review]

File: test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js (hunk 1 above)
Comment (leobalter) [reply test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js:36]: Yes! This error happens by the time of the evaluation and somehow I got it as throwing when calling the fn.

Comment (leobalter) [reply test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js:36]: @rwaldron I believe your initial review is actually correct.

```
CopyNameAndLength ( F: a function object, Target: a function object, optional prefix: a String, optional argCount: a Number, )
1. If argCount is undefined, then set argCount to 0.
2. Let L be 0.
3. Let targetHasLength be ? HasOwnProperty(Target, "length").
4. If targetHasLength is true, then
  a. Let targetLen be ? Get(Target, "length").
``

The abrupt completion happens during the evaluation time when creating a wrapped function for `fn` and running Get(fn, "length")

In this test, `r.evaluate()` is the throwing factor. This also leads to some missing coverage:

- tests for the `HasOwnProperty(length)` part (e.g. callable proxies), including abrupt completion.
- tests for `length` in the prototype chain (mixed with the hasOwn internal)
- resolution of name/length for values evaluated from other wrapped functions (e.g. a return from function, or sending values back to the shadowRealm).

File: test/built-ins/ShadowRealm/WrappedFunction/name.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  configurable: true,
+});
+fn;
+`);
+verifyProperty(wrapped, "name", {
+  value: "",
+  enumerable: false,
+  writable: false,
+  configurable: true,
+});
```
[the remark this answers is not in the document]
Comment (legendecas) [reply test/built-ins/ShadowRealm/WrappedFunction/name.js:80]: Submitted https://github.com/tc39/test262/pull/3396 :)

File: test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js (hunk 1 above)
Comment (legendecas) [reply test/built-ins/ShadowRealm/WrappedFunction/length-throws-typeerror.js:36]: Yeah, @rwaldron is right. Submitted https://github.com/tc39/test262/pull/3396 to fix this.

**Pull request merged**